File: php/features.php
<?php
declare(strict_types=1);

// IssCurrentLocation SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IssCurrentLocationFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IssCurrentLocationBaseFeature();
            case "test":
                return new IssCurrentLocationTestFeature();
            default:
                return new IssCurrentLocationBaseFeature();
        }
    }
}

File: py/features.py
# IssCurrentLocation SDK feature factory

from feature.base_feature import IssCurrentLocationBaseFeature
from feature.test_feature import IssCurrentLocationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IssCurrentLocationBaseFeature(),
        "test": lambda: IssCurrentLocationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
